libs/sdk-ui-kit/src/Dropdown/Dropdown.tsx

```diff
@@ -136,38 +153,46 @@ export const Dropdown: React.FC<IDropdownProps> = (props) => {
         enableEventPropagation = false,
         closeOnEscape = false,
         autofocusOnOpen = false,
+
+        accessibilityConfig,
     } = props;
-    const [{ isOpen }, setState] = useState<IDropdownState>({
-        isOpen: !!openOnInit,
-    });
+    const [isOpen, setIsOpen] = usePropState(isOpenProp ?? openOnInit ?? false);
 
     const id = useId();
     const dropdownId = `dropdown-${id}`;
+    const dropdownButtonId = `dropdown-button-${id}`;
 
+    const buttonWrapperRef = useRef<HTMLElement | null>(null);
     const _renderButton = (renderProps: IDropdownButtonRenderProps) => (
-        <div className={dropdownId}>{renderButton(renderProps)}</div>
+        <DropdownButtonKeyboardWrapper
+            onToggle={renderProps.toggleDropdown}
+            isOpen={renderProps.isOpen}
+            ref={buttonWrapperRef}
+            id={dropdownButtonId}
+        >
+            {renderButton(renderProps)}
+        </DropdownButtonKeyboardWrapper>
     );
 
-    const toggleDropdown = useCallback((): void => {
-        setState((state) => ({
-            ...state,
-            isOpen: !state.isOpen,
-        }));
-    }, []);
+    const toggleDropdown = useCallback(
+        (desiredState?: boolean | unknown): void => {
+            if (typeof desiredState === "boolean") {
+                onToggle ? onToggle(desiredState) : setIsOpen(desiredState);
+            } else {
+                onToggle ? onToggle() : setIsOpen((state) => !state);
+            }
+        },
+        [onToggle, setIsOpen],
+    );
 
     const closeDropdown = useCallback((): void => {
-        setState((state) => ({
-            ...state,
-            isOpen: false,
-        }));
-    }, []);
+        toggleDropdown(false);
+        (buttonRef.current ? buttonRef : buttonWrapperRef).current?.focus();
```

Comment:
return of focus to trigger is handled in UiFocusTrap too. Keep it only on one of these two places